wordpress-mobile/WordPress-Android
# [Reader Improvements] Update follow/unfollow button on site header and tag header

### Do not merge
~Waiting for discussion: p1695428757161569-slack-C05N140C8H5~ Discussion will be handled in a separate issue.

Part of #19078 

This PR updates the design of follow/unfollow button for site header and tag header. The post header and "Sites to follow" design updates will be done in separate PRs.

Design specs: zQnohyMpLzBzQ5jzMMKni3-fi


To test:
1 - Run JP and sign in;
2 - Open debug settings and enable `ReaderImprovementsFeatureConfig` FF;
3 - Open reader;
4 - Tap on any site: see if the follow/unfollow button matches the design. Tap on it to change it's status and make sure the follow action is working as expected;
5 - Tap on any tag: check the same same as step 4;
6 - Disable the FF that was enabled on step 2;
7 - Open reader;
8 - Tap on any site: follow/unfollow button design should be the same, but the site header layout should be the previous one (before Reader Improvements);
9 - Tap on any tag: same as step 8, but with the tag header layout.

### Site header (FF enabled)
<p float="left">
<img width="300" alt="Screenshot 2023-09-22 at 9 47 51 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/6af9eb42-23f5-4cf6-8eed-524cf7b18b70">
<img width="300" alt="Screenshot 2023-09-22 at 9 47 58 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/ff81e441-1737-4d3a-b83d-a99ecd5c2237">
<img width="300" alt="Screenshot 2023-09-22 at 9 48 17 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/fb300daa-5b6e-4772-b86d-a91a83bf175c">
<img width="300" alt="Screenshot 2023-09-22 at 9 48 14 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/fff4e4ef-dc46-408f-afec-e327a5399a03">
</p>

### Tag header (FF enabled)
<p float="left">
<img width="300" alt="Screenshot 2023-09-22 at 9 49 43 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/bbd7ffe0-3124-4a2a-85be-9e70b55fab2c">
<img width="300" alt="Screenshot 2023-09-22 at 9 49 36 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/1149e0d5-99ad-4c6d-b51d-8352e928e638">
<img width="300" alt="Screenshot 2023-09-22 at 9 49 17 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/3d17fdd8-e46d-4216-8500-337d8f12ba5d">
<img width="300" alt="Screenshot 2023-09-22 at 9 49 23 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/dd6f86cf-7c4a-4982-aa64-497da5212f92">

</p>

### Site header (FF disabled)
<p float="left">
<img width="300" alt="Screenshot 2023-09-22 at 9 44 05 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/6bf6856d-2c1b-4e5d-bcae-b8839c9aebab">
<img width="300" alt="Screenshot 2023-09-22 at 9 44 10 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/29eb0608-2f2c-4922-8c08-14c878388087">
<img width="300" alt="Screenshot 2023-09-22 at 9 51 58 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/ebc5ea5d-7ebf-435c-acda-01ccdf34afc7">
<img width="300" alt="Screenshot 2023-09-22 at 9 52 22 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/2e36b94d-0e95-407e-aa6c-c06b2e1f85aa">
</p>

### Tag header (FF disabled)
<p float="left">
<img width="300" alt="Screenshot 2023-09-22 at 9 45 20 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/4ba3f8b5-22b5-447d-8071-c3159663492b">
<img width="300" alt="Screenshot 2023-09-22 at 9 45 27 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/bd5a9926-10ea-44ba-a73a-5a82eef312bc">
<img width="300" alt="Screenshot 2023-09-22 at 9 53 21 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/34b84b8e-aaa7-4275-afd6-d2cc2e2fdd20">
<img width="300" alt="Screenshot 2023-09-22 at 9 53 14 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/06beaed5-d643-4196-9688-adf51c955a27">
</p>


## Regression Notes
1. Potential unintended areas of impact
Other places I might've overlooked that are using the same follow button style

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Changes were only made to views

PR submission checklist:


UI Changes testing checklist:
